base_layer/p2p/src/dht_service/dht_service.rs

```diff
@@ -88,52 +88,89 @@ impl DHTService {
     }
 
     /// Construct a new join message that contains the current nodes identity and net_addresses
-    fn construct_join_msg(&self) -> JoinMessage {
-        JoinMessage {
-            node_id: self.node_id.clone(),
-            public_key: self.public_key.clone(),
-            net_address: vec![self.control_service_address.clone()],
-        }
+    fn construct_join_msg(&self) -> Result<JoinMessage, DHTError> {
+        let node_identity = self.node_identity.clone().ok_or(DHTError::NodeIdentityUndefined)?;
+        Ok(JoinMessage {
+            node_id: node_identity.identity.node_id.clone(),
+            net_address: vec![node_identity.control_service_address.clone()],
+        })
+    }
+
+    /// Construct a new discover message that contains the current nodes identity and net_addresses
+    fn construct_discover_msg(&self) -> Result<DiscoverMessage, DHTError> {
+        let node_identity = self.node_identity.clone().ok_or(DHTError::NodeIdentityUndefined)?;
+        Ok(DiscoverMessage {
+            node_id: node_identity.identity.node_id.clone(),
+            net_address: vec![node_identity.control_service_address.clone()],
+        })
     }
 
     /// Send a new network join request to the peers that are closest to the current nodes network location. The Join
     /// Request will allow other peers to be able to find this node on the network.
     fn send_join(&self) -> Result<(), DHTError> {
         let oms = self.oms.clone().ok_or(DHTError::OMSUndefined)?;
+        let node_identity = self.node_identity.clone().ok_or(DHTError::NodeIdentityUndefined)?;
 
         oms.send_message(
             BroadcastStrategy::Closest(ClosestRequest {
                 n: DHT_BROADCAST_NODE_COUNT,
-                node_id: self.node_id.clone(),
+                node_id: node_identity.identity.node_id.clone(),
                 excluded_peers: Vec::new(),
             }),
             MessageFlags::NONE,
-            self.construct_join_msg(),
+            self.construct_join_msg()?,
         )?;
         trace!(target: LOG_TARGET, "Join Request Sent");
 
         Ok(())
     }
 
-    /// Send a network join update request to a specific peer to update contact information.
-    fn send_join_direct(&self, node_id: &NodeId) -> Result<(), DHTError> {
+    /// Send a network join update request directly to a specific known peer
+    fn send_join_direct(&self, dest_public_key: CommsPublicKey) -> Result<(), DHTError> {
         let oms = self.oms.clone().ok_or(DHTError::OMSUndefined)?;
 
         oms.send_message(
-            BroadcastStrategy::DirectNodeId(node_id.clone()),
+            BroadcastStrategy::DirectPublicKey(dest_public_key.clone()),
```

Comment:
Shouldn't need to clone

Reply:
Good catch, was an artefact from the bug hunting. Change applied